metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala

```diff
@@ -80,29 +79,25 @@ class SemanticdbIndexer(
   private def onChange(path: AbsolutePath, docs: TextDocuments): Unit =
     providers.foreach(_.onChange(docs, path))
 
-  def onChange(file: Path): Unit = {
-    if (!Files.isDirectory(file)) {
-      if (file.isSemanticdb) {
-        try {
-          val docs = TextDocuments.parseFrom(Files.readAllBytes(file))
-          SemanticdbClasspath.toScala(workspace, AbsolutePath(file)).foreach {
-            sourceFile => onChange(sourceFile, docs)
-          }
-        } catch {
-          /* @note in some cases file might be created or modified, but not actually finished
-           * being written. In that case, exception here is expected and a new event will
-           * follow after it was finished.
-           */
-          case e: InvalidProtocolBufferException =>
-            scribe.debug(s"$file is not yet ready", e)
-          /* @note FileSystemException is thrown on Windows instead of InvalidProtocolBufferException */
-          case e: FileSystemException =>
-            scribe.debug(s"$file is not yet ready", e)
-          case NonFatal(e) =>
-            scribe.warn(s"unexpected error processing the file $file", e)
+  def onChange(file: SemanticdbPath): Unit = {
+    if (!Files.isDirectory(file.toNIO)) {
```

Comment:
Is 
```
In case the path is not null, we walk up the file tree to the parent `META-INF/semanticdb`
parent directory and re-index all of its `*.semanticdb` children.

In case of a null path, we re-index `META-INF/semanticdb` for all targets
```
true taking into account that there was and still is check `if (!Files.isDirectory(file.toNIO))`. Anything which wasn't a valid semanticDb file was being discarded by this function. That's why I left `// stream.forEach(onChange(_))` commented, I don't know what needs to be done and I'm a bit confused.

Reply:
ok, stream is a stream of files in directory so they will be semanticDB files